astropy/table/tests/test_mixin.py

```diff
@@ -118,7 +118,7 @@ def test_io_quantity_write(tmpdir):
     open(filename, 'w').close()
 
     for fmt in ('fits', 'votable', 'hdf5'):
-        if fmt == 'fits':
+        if fmt != 'hdf5':
```

Comment:
I usually advocate minimal changes, but in this case the logic here has gotten really tortured for no good reason.   Suggest:
```
    # Show that FITS and VOTable formats succeed
    for fmt in ('fits', 'votable'):
        t.write(filename, format=fmt, overwrite=True)
        qt = QTable.read(filename, format=fmt)
        assert isinstance(qt['a'], u.Quantity)
        assert qt['a'].unit == 'Angstrom'

    # Show that HDF5 format fails
    if HAS_H5PY:
         with pytest.raises(ValueError) as err:
             t.write(filename, format='hdf5', overwrite=True)
         assert 'cannot write table with mixin column(s)' in str(err.value)
```